### PR TITLE
Sns helpers

### PR DESCRIPTION
**Pull request by markbulgakov**

### CODE REVIEW

File: sosw/components/helpers.py [hunk 1]
```diff
@@ -887,3 +890,33 @@ def to_bool(val):
         elif val.lower() in ['false', '0']:
             return False
     raise Exception(f"Can't convert unexpected value to bool: {val}, type: {type(val)}")
+
+
+def get_message_dict_from_sns_event(event):
+    """
+    Extract SNS event message and returns it as dict
+
+    :param dict event: SNS Event (payload)
```
Comment (nimirium) [inline sosw/components/helpers.py]: I would say
`:param dict event: Lambda SNS event (payload)`
`:return: The SNS message, converted to dict`

File: sosw/components/helpers.py [hunk 2]
```diff
@@ -887,3 +890,33 @@ def to_bool(val):
...
+    """
+
+    message = event['Records'][0]['Sns'].get('Message')
+
+    return json.loads(message)
+
+
+def is_event_from_sns(event):
+    """
+    Check if the invocation was with SNS.
```
Comment (nimirium) [inline sosw/components/helpers.py]: "Check if the **lambda** invocation was with SNS."

File: sosw/components/helpers.py [hunk 3]
```diff
@@ -887,3 +890,33 @@ def to_bool(val):
...
+
+    :param dict event: SNS / Lambda Event (payload)
```
Comment (nimirium) [inline sosw/components/helpers.py]: `:param dict event: Lambda Event (payload)`

File: sosw/components/helpers.py [hunk 4]
```diff
@@ -887,3 +890,33 @@ def to_bool(val):
...
+
+    :param dict event: Lambda SNS event (payload)
+
+    :rtype dict
+    :return: The SNS message, converted to dict
+    """
+
+    message = event['Records'][0]['Sns'].get('Message')
+
+    return json.loads(message)
```
Comment (ngr) [inline sosw/components/helpers.py]: json.loads(None) - raise
json.loads("") - raise...
pretty unstable behaviour. You probably want to catch these ones at least are return empty dict.

File: sosw/components/test/unit/test_helpers.py [hunk 5]
```diff
@@ -666,5 +666,23 @@ def test_to_bool(self):
...
+        event_empty = {'Records': [{'Sns': {}}]}
+
+        self.assertEqual(get_message_dict_from_sns_event(event_true), {})
+        self.assertRaises(ValueError, validate_account_to_dashed, event_false)
+        self.assertRaises(ValueError, validate_account_to_dashed, event_empty)
+
+
+    def test_is_event_from_sns(self):
+        event_true = {'Records': [{'Sns': {'Message': ''}}]}
+        event_false = {'Records': [{}]}
```
Comment (ngr) [inline sosw/components/test/unit/test_helpers.py]: Make sure to handle these please:
event_false = ""
event_false = {}
event_false = "{}"

File: sosw/components/test/unit/test_helpers.py [hunk 6]
```diff
@@ -666,5 +666,23 @@ def test_to_bool(self):
             to_bool(object())
 
 
+    def test_get_message_dict_from_sns_event(self):
+        event_true = {'Records': [{'Sns': {'Message': '{}'}}]}
+        event_false = {'Records': [{'Sns': {'Message': ''}}]}
+        event_empty = {'Records': [{'Sns': {}}]}
+
+        self.assertEqual(get_message_dict_from_sns_event(event_true), {})
+        self.assertRaises(ValueError, validate_account_to_dashed, event_false)
```
Comment (ngr) [inline sosw/components/test/unit/test_helpers.py]: `validate_account_to_dashed` has nothing to do here. Please update the tests appropriately and have an easy way to add more edge-cases please. TESTS_TRUE = {...}, TESTS_FALSE = {...} etc
